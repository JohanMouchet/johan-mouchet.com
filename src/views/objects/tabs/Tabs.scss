@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.o-tabs {
  $root: &;

  margin: t-vertical-rhythm(1) 0;

  /* Thumbs */
  &__thumbs {
    display: flex;
    padding-left: 0;
    border-bottom: 1px solid $s-color-gray;
    list-style: none;
    overflow-x: auto;
    overflow-y: hidden;
  }

  &__thumb {
    flex: 0 1 auto;
    white-space: nowrap;
    position: relative;

    /** Active **/
    &.is--active #{$root}__button {
      color: $s-color-link;
      font-weight: 700;

      &:hover,
      &:focus {
        cursor: default;
      }

      &::after {
        opacity: 1;
        transform: scaleX(1);
      }
    }
  }

  /* Button */
  &__button {
    background-color: transparent;
    border: 0;
    outline: 0;
    all: unset;

    @include t-link($s-color-text-dimmed, "none", $hover-decoration: "none");

    display: block;
    padding: t-vertical-rhythm(0.5) 1.5em;
    cursor: pointer;

    &::after {
      content: "";
      position: absolute;
      bottom: 0;
      left: 0;
      height: 2px;
      width: 100%;
      background-color: currentColor;
      opacity: 0;
      transform: scaleX(0);
      transform-origin: 50% 0;
      transition: opacity 0.2s ease-out, transform 0.2s ease-out;
    }
  }

  /* Panels */
  &__panels {
    padding-left: 0;
    list-style: none;
  }

  &__panel {
    display: none;
  }
}
