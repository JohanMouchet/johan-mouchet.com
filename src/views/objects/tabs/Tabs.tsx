import React from "react";
import cx from "classnames";
import "./Tabs.scss";

type Props = {
  tabs: Array<{
    thumb: string;
    panel: string;
  }>;
  className?: string | string[] | { [key: string]: boolean };
};

const Tabs: React.FC<Props> = ({ tabs, className }) => {
  const classNames = cx("o-tabs", className);

  return !tabs?.length ? null : (
    <div className={classNames}>
      <ul className="o-tabs__thumbs">
        {tabs.map((tab, index) => (
          <li
            className="o-tabs__thumb u-animation u-animation--fade-in-up"
            style={{
              animationDelay: index > 0 ? index * 0.15 + "s" : undefined,
            }}
            key={tab.thumb}
          >
            <button className="o-tabs__button" type="button">
              {tab.thumb}
            </button>
          </li>
        ))}
      </ul>
      <ul className="o-tabs__panels">
        {tabs.map((tab, index) => (
          <li className="o-tabs__panel" key={index}>
            {tab.panel}
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Tabs;
