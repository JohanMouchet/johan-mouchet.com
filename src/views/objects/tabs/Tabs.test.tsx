import React from "react";
import { render } from "@testing-library/react";
import Tabs from ".";

it("Renders <Tabs /> without crashing", () => {
  expect(
    render(
      <Tabs
        tabs={[
          {
            thumb: "Thumb",
            panel: "Panel",
          },
        ]}
      />
    ).container.firstChild
  ).toMatchSnapshot();
});
