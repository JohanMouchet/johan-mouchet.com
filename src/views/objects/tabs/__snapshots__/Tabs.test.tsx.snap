// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Tabs /> without crashing 1`] = `""`;
